core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -2470,6 +2478,10 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
   final case class AnyRefArray[A <: AnyRef](array: Array[A], offset: Int, override val length: Int)
       extends Arr[A]
       with ChunkIterator[A] { self =>
+
+    @threadUnsafe
+    lazy val classTag: ClassTag[A] = Tags.fromValue(array(0))
```

Comment:
The previous implementation was inherited from `Arr` and was:
```scala
    implicit val classTag: ClassTag[A] =
      ClassTag(array.getClass.getComponentType)
```

This new implemetation is inspired from the one in `VectorChunk`

Not sure which one to prefer